lib/active_merchant/billing/gateways/paymentez.rb

```diff
@@ -0,0 +1,274 @@
+        post[:order][:discount] = options[:discount] if options[:discount]
+        post[:order][:installments] = options[:installments] if options[:installments]
+        post[:order][:installments_type] = options[:installments_type] if options[:installments_type]
+        post[:order][:taxable_amount] = options[:taxable_amount] if options[:taxable_amount]
+      end
+
+      def add_payment(post, payment)
+        post[:card] ||= {}
+        post[:card][:number] = payment.number
+        post[:card][:holder_name] = payment.name
+        post[:card][:expiry_month] = payment.month
+        post[:card][:expiry_year] = payment.year
+        post[:card][:cvc] = payment.verification_value
+        post[:card][:type] = CARD_MAPPING[payment.brand]
+      end
+
+      def parse(body)
+        JSON.parse(body)
+      end
+
+      def commit_raw(object, action, parameters)
+        url = "#{(test? ? test_url : live_url)}#{object}/#{action}"
+
+        begin
+          raw_response = ssl_post(url, post_data(parameters), headers)
+        rescue ResponseError => e
+          raw_response = e.response.body
+        end
+        parse(raw_response)
+      end
+
+      def commit_transaction(action, parameters)
+        response = commit_raw('transaction', action, parameters)
+        Response.new(
+          success_from(response),
+          message_from(response),
+          response,
+          authorization: authorization_from(response),
+          test: test?,
+          error_code: error_code_from(response)
+        )
+      end
+
+      def commit_card(action, parameters)
+        response = commit_raw('card', action, parameters)
+        Response.new(
+          card_success_from(response),
+          card_message_from(response),
+          response,
+          authorization: card_authorization_from(response),
+          test: test?,
+          error_code: card_error_code_from(response)
+        )
+      end
+
+      def headers
+        {
+          'Auth-Token' => authentication_code,
+          'Content-Type' => 'application/json'
+        }
+      end
+
+      def success_from(response)
+        !response.include?('error') && (response['status'] || response['transaction']['status']) == 'success'
+      end
+
+      def card_success_from(response)
+        return false if response.include?('error')
+        return true if response['message'] == 'card deleted'
+        response['card']['status'] == 'valid'
+      end
+
+      def message_from(response)
+        if success_from(response)
+          (response['transaction'] || {})['message']
+        else
+          (response['error'] || {})['type']
+        end
+      end
+
+      def card_message_from(response)
+        if !response.include?('error')
+          response['message'] || response['card']['message']
+        else
+          response['error']['type']
+        end
+      end
+
+      def authorization_from(response)
+        (response['transaction'] || {})['id']
```

Comment:
@dtykocki For my own edification as well, since I don't think I've seen it before, do you have an opinion on this pattern vs. something like a more conventional verbose
`response['transaction']['id'] if response['transaction'] && response['transaction']['id']`?

Reply:
It's for sure less verbose, which is pretty nice, but I'd prefer to stick with our current convention for consistency sake.

Reply:
The dying Pythonista in me respects but is saddened by your choice, @dtykocki ;-). I'll switch this pattern though.

Reply:
Well.. if Ruby had pattern matching like Elixir, this would be a nil point.

Reply:
Seriously, @dtykocki, I was just dying for F#/Elixir/Haskell pattern matching throughout this PR. Given the heterogeneous nature of their JSON, that'd have simplified things _so much_ it's not funny.